src/inputs/pcap/PcapInputStream.cpp

```diff
@@ -229,30 +229,44 @@ void PcapInputStream::stop()
 
 void PcapInputStream::tcp_message_ready(int8_t side, const pcpp::TcpStreamData &tcpData)
 {
-    tcp_message_ready_signal(side, tcpData);
+    std::unique_lock lock(_input_mutex);
```

Comment:
ouch, it sucks to take a mutex lock hit here per event. we can make this shared_lock instead of unique lock though, since we are not mutating _callbacks. i think we'll need to benchmark this.

Reply:
shared for those calls did improve the calls. did not see performance loss between current implementation and new implementation